pkg/golinters/godoclint/godoclint.go

```diff
@@ -9,26 +9,41 @@ import (
 )
 
 func New(settings *config.GodoclintSettings) *goanalysis.Linter {
-	pcfg := glconfig.PlainConfig{
-		Include: settings.Include,
-		Exclude: settings.Exclude,
-		Enable:  settings.Enable,
-		Disable: settings.Disable,
-		Options: &glconfig.PlainRuleOptions{
-			MaxLenLength:                   settings.Options.MaxLen.Length,
-			MaxLenIncludeTests:             settings.Options.MaxLen.IncludeTests,
-			PkgDocStartWith:                settings.Options.PkgDoc.StartWith,
-			PkgDocIncludeTests:             settings.Options.PkgDoc.IncludeTests,
-			SinglePkgDocIncludeTests:       settings.Options.SinglePkgDoc.IncludeTests,
-			RequirePkgDocIncludeTests:      settings.Options.RequirePkgDoc.IncludeTests,
-			RequireDocIncludeTests:         settings.Options.RequireDoc.IncludeTests,
-			RequireDocIgnoreExported:       settings.Options.RequireDoc.IgnoreExported,
-			RequireDocIgnoreUnexported:     settings.Options.RequireDoc.IgnoreUnexported,
-			StartWithNamePattern:           settings.Options.StartWithName.Pattern,
-			StartWithNameIncludeTests:      settings.Options.StartWithName.IncludeTests,
-			StartWithNameIncludeUnexported: settings.Options.StartWithName.IncludeUnexported,
-			NoUnusedLinkIncludeTests:       settings.Options.NoUnusedLink.IncludeTests,
-		},
+	var pcfg glconfig.PlainConfig
+
+	if settings != nil {
+		// The following options are explicitly ignored: they must be handled globally with exclusions or nolint directives.
+		// - Include
+		// - Exclude
+
+		// The following options are explicitly ignored: these options cannot work as expected because the global configuration about tests.
+		// - Options.MaxLenIncludeTests
+		// - Options.PkgDocIncludeTests
+		// - Options.SinglePkgDocIncludeTests
+		// - Options.RequirePkgDocIncludeTests
+		// - Options.RequireDocIncludeTests
+		// - Options.StartWithNameIncludeTests
+		// - Options.NoUnusedLinkIncludeTests
+
+		pcfg = glconfig.PlainConfig{
+			Enable:  settings.Enable,
+			Disable: settings.Disable,
+			Options: &glconfig.PlainRuleOptions{
+				MaxLenLength:                   settings.Options.MaxLen.Length,
+				MaxLenIncludeTests:             pointer(true),
```

Comment:
@ldez I think `IncludeTests` options should be set to `false`, as this is the default behaviour of the linter. Correct me if I'm missing sth.

Reply:
The values don't depend on the default values of the linter, but rather on the possibility to manage them globally.

Reply:
But it's now set to `true` and that tells the linter to always include test files, which is a rare case in practice. So, if we're going to drop the option from config, it's better to populate it with `false`.

Reply:
I think it's better for consistency between godocs to enable this option by default.

The test files can be excluded with the global configuration:
```yml
linters:
  exclusions:
    rules:
      - path:  (.+)_test\.go
        text: godoc exceeds max length
        linters:
          - godoclint
```